src/app/frontend/common/services/global/kubeconfig.ts

```diff
@@ -0,0 +1,246 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Inject, Injectable, NgZone} from '@angular/core';
+import {HttpErrorResponse} from '@angular/common/http';
+import {Router} from '@angular/router';
+import {CookieService} from 'ngx-cookie-service';
+import {SessionStorageService} from 'ngx-webstorage';
+import {load as loadYaml, dump as dumpYaml} from 'js-yaml';
+
+import {AsKdError, K8SError} from '../../errors/errors';
+import {AuthService} from './authentication';
+import {PluginsConfigService} from './plugin';
+import {LoginSpec} from '@api/backendapi';
+import {KdError} from '@api/frontendapi';
+import {Config, CONFIG_DI_TOKEN} from '../../../index.config';
+
+type Cluster = {
+  name: string;
+  cluster: {
+    server: string;
+    'certificate-authority-data': string;
+    'sidecar-host': string;
+  };
+};
+type User = {
+  name: string;
+  user: {
+    token: string;
```

Comment:
We can't store token here as it is already part of our login mechanism and is stored in an encrypted cookie. This would be extremely insecure. Also, we do not support `client-certificate-data` as a way of logging in anyway, so this is obsolete.

Reply:
If we store all the tokens contained in kubeconfig in cookies, I'm worried that all HTTP requests will get too big and impact performance.
I'm considering to encrypt tokens in jwe and store them into cookie before login sequence. But do you have any other ideas?